mithril-client/src/aggregator_client/mithril_stake_distribution_client.rs

```diff
@@ -40,9 +44,9 @@ impl MithrilStakeDistributionClient {
 
         match self.http_client.get_content(&url).await {
             Ok(content) => {
-                let message: MithrilStakeDistributionMessage = serde_json::from_str(&content)?;
+                let message: MithrilStakeDistributionMessage = serde_json::from_str(&content).with_context(|| format!("MithrilStakeDistribution Client can not deserialize artifact at '{url}'"))?;
                 let stake_distribution_entity =
-                    FromMithrilStakeDistributionMessageAdapter::try_adapt(message)?;
+                    FromMithrilStakeDistributionMessageAdapter::try_adapt(message).with_context(|| format!("MithrilStakeDistribution Client can not convert message for hash '{hash:?}'"))?;
```

Comment:
I do not understand the error message. Maybe a proper error message would be `Cannot parse stake distribution artifact from message. (certificate hash='{hash}')`. WDYT?

Reply:
It seems clearer and more precise indeed.

Reply:
Should I specify "snapshot" or "stake distribution" before each "artifact" in the other error messages?

Reply:
We already know which artifact it is when the command is executed, so maybe it would be lighter not to specify the type in the error message?